tree/ntuple/v7/inc/ROOT/RField/RFieldFundamental.hxx

```diff
@@ -346,72 +346,164 @@ public:
 /// Template specializations for floating-point types
 ////////////////////////////////////////////////////////////////////////////////
 
+extern template class RSimpleField<double>;
 extern template class RSimpleField<float>;
 
-template <>
-class RField<float> final : public RSimpleField<float> {
-   std::size_t fBitWidth = sizeof(float) * 8;
-   double fValueMin = std::numeric_limits<float>::min();
-   double fValueMax = std::numeric_limits<float>::max();
+template <typename T>
+class RFloatField : public RSimpleField<T> {
```

Comment:
```suggestion
class RRealField : public RSimpleField<T> {
```
to avoid the 'seemingly' redundant `RFloatField<float>` :) and because we seem to use `Real` in other places for similar purposes.